Prog/TOPA/TOPA_add_subsidies.sas

```diff
@@ -0,0 +1,169 @@
+
+data TOPA_subsidy; 
+  set TOPA_subsidy_match; 
+  format all_POA_start MMDDYY10.;
+  format actual_POA_start DYESNO.;
+  format date_LEC_form MMDDYY10.;
+  if not(missing(POA_start_orig)) then do;
+	all_POA_start=POA_start_orig;
+	actual_POA_start=1;
+		end;
+  else if (missing(POA_start_orig)) then do; 
+    all_POA_start=POA_start_orig;
+	actual_POA_start=0;
+		end;
+  label all_POA_start='Period of affordability, original start date or if original date missing, period of affordability, current start date';
+  label actual_POA_start='Original start date used';
+  u_days_notice_to_subsidy = all_POA_start - u_sale_date;
+  label u_days_notice_to_subsidy='Number of days from notice of sale to start of subsidy (Urban created var)';
+  if not(missing(all_POA_start)); ** delete missing subsidy start dates **; 
+  if portfolio in ("LIHTC", "202/811", "PB8", "PRAC", "DC HPTF", "LECOOP"); ** only include LIHTC, federal project-based subsidies, DC HPTF, and LEC **;
+  if portfolio = "LIHTC" and u_days_notice_to_subsidy < 0 then before_LIHTC_aff_units=Units_Assist;
+  if portfolio = "LIHTC" and 0 <= u_days_notice_to_subsidy <= 365 then after_LIHTC_aff_units=Units_Assist;
+  if portfolio in ( "202/811", "PB8", "PRAC" ) and u_days_notice_to_subsidy < 0 then before_fed_aff_units=Units_Assist;
+  if portfolio in ( "202/811", "PB8", "PRAC" ) and 0 <= u_days_notice_to_subsidy <= 365 then after_fed_aff_units=Units_Assist;
+  if portfolio = "DC HPTF" and u_days_notice_to_subsidy < 0 then before_DC_HPTF_aff_units=Units_Assist;
+  if portfolio = "DC HPTF" and 0 <= u_days_notice_to_subsidy <= 365 then after_DC_HPTF_aff_units=Units_Assist;
+  if portfolio = "LECOOP" and u_days_notice_to_subsidy < 0 then before_LEC_aff_units=Units_Assist;
+  if portfolio = "LECOOP" and 0 <= u_days_notice_to_subsidy <= 365 then after_LEC_aff_units=Units_Assist;
+  if missing(poa_end_actual) or u_sale_date-poa_end_actual > 365 or poa_end_actual > '01Jan2023'd; **not including units where subsidy's expired **;
+run;
+
+%File_info( data=TOPA_subsidy, printobs=10 )
+
+**Aggregating the data across rows with the same ID **;
+
+proc summary data=TOPA_subsidy
+ noprint;
+ var before_LIHTC_aff_units after_LIHTC_aff_units before_fed_aff_units after_fed_aff_units before_DC_HPTF_aff_units 
+	after_DC_HPTF_aff_units before_LEC_aff_units after_LEC_aff_units;
+ class id;
+ output out=TOPA_sum_rows
+	sum=sum_before_LIHTC_aff_units sum_after_LIHTC_aff_units sum_before_fed_aff_units sum_after_fed_aff_units sum_before_DC_HPTF_aff_units
+	sum_after_DC_HPTF_aff_units sum_before_LEC_aff_units sum_after_LEC_aff_units;
+run;
+
+%File_info( data=TOPA_sum_rows, printobs=10)
+
+**Summing the data across columns by subsidy type**;
+
+data TOPA_sum_total; 
+  set TOPA_sum_rows; 
+  if not(missing(sum_before_LIHTC_aff_units)) and not(missing(sum_after_LIHTC_aff_units)) then total_LIHTC_aff_units=sum_before_LIHTC_aff_units+sum_after_LIHTC_aff_units;
```

Comment:
I'm not sure why you are creating these totals. There's no reason to add the before and after numbers together and, in fact, that sum isn't meaningful because some units are going to be both before and after, so we'd be double counting.